services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/geometria/HankeGeometriatDaoImpl.kt

```diff
@@ -185,4 +152,68 @@ class HankeGeometriatDaoImpl(private val jdbcOperations: JdbcOperations) : Hanke
             }, hankeGeometriatId
         )
     }
+
+    override fun updateHankeGeometriat(hankeGeometriat: HankeGeometriat) {
+        with(jdbcOperations) {
+            update(
+                """
+                UPDATE HankeGeometriat
+                SET
+                    version = ?,
+                    modifiedByUserId = ?,
+                    modifiedAt = ?
+                WHERE
+                    id = ?
+            """.trimIndent()
+            ) { ps ->
+                ps.setInt(1, hankeGeometriat.version!!)
+                if (hankeGeometriat.modifiedByUserId != null) {
+                    ps.setString(2, hankeGeometriat.modifiedByUserId!!)
+                } else {
+                    ps.setNull(2, Types.INTEGER)
+                }
+                if (hankeGeometriat.modifiedAt != null) {
+                    ps.setTimestamp(3, Timestamp(hankeGeometriat.modifiedAt!!.toInstant().toEpochMilli()))
+                } else {
+                    ps.setNull(3, Types.TIMESTAMP)
+                }
+                ps.setInt(4, hankeGeometriat.id!!)
+            }
+            // delete old geometry rows
+            deleteHankeGeometriaRows(hankeGeometriat, this)
+            // save new geometry rows
+            saveHankeGeometriaRows(hankeGeometriat, this)
+        }
+    }
+
```

Comment:
Almost fully identical code; updateHankeGeometriat only adds a call to saveHankeGeometriaRows() in the end.. perhaps call deleteHankeGeometriat() from updateHankeGeometriat to avoid duplicate code?